crates/cubecl-wgpu/src/compute/server.rs

```diff
@@ -196,11 +194,21 @@ where
         &mut self,
         binding: server::Binding<Self>,
     ) -> <Self::Storage as cubecl_runtime::storage::ComputeStorage>::Resource {
-        self.memory_management.get_resource(
-            binding.memory,
-            binding.offset_start,
-            binding.offset_end,
-        )
+        // Keep track of any buffer that might be used in the wgpu queue, as we cannot copy into them
+        // after they have any outstanding compute work. Calling get_resource repeatedly
+        // will add duplicates to this, but that is ok.
+        let handle = self.memory_management.get(binding.memory.clone());
+        self.storage_in_flight.push(handle.id);
```

Comment:
We could remove duplicates by using a set? If the `storage_in_flight` is cleared often it's probably not worth it.

Reply:
I think we should actually probably create a data structure with methods. We use a slice in the memory management functions, but it could be a struct with the proper methods. Then we could change the data structure more easily (Vec under size of 10-20, a set if more than that).